oras/auth/__init__.py

```diff
@@ -14,10 +14,11 @@ class AuthenticationException(Exception):
     pass
 
 
-def get_auth_backend(name="token", session=None, **kwargs):
+def get_auth_backend(name="token", session=None, prefix="https", **kwargs):
```

Comment:
step 2.

unless otherwise specified, this function `get_auth_backend()` parameter `prefix` defaults to https; but for all usages in oras code, is set from provider's prefix, as we seen in step1.